### PR TITLE
projectile graph hwk

### PR DESCRIPTION
**Pull request by cblitz2**
Screenshot 2024-02-28 at 4 27 18 PM

### CODE REVIEW

**Comment by andrewoid** [comment]
What inputs does it fail with. Can you post a screenshot of what it looks like when it fails?

File: src/main/java/blitz/projectile/ProjectileGraph.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+        for (int i = 0; i < projectile.getApexTime() * 2; i++) {
+            currX = (int) projectile.getX();
+            currY = (int) projectile.getY();
+            projectile.setTime(i);
+            g.drawLine(currX, -currY, (int) projectile.getX(), -(int) projectile.getY());
+        }
+
+        g.setColor(Color.BLUE);
+        g.fillOval((int) projectile.getX() / 2, -(int) projectile.getPeakY(), 10, 10);
```
Comment (andrewoid) [inline src/main/java/blitz/projectile/ProjectileGraph.java]: Fill oval doesn't draw an oval with its center at x,y, it draws an oval where it's CORNER is x,y

**Comment by cblitz2** [comment]
Screenshot 2024-02-27 at 10 37 12 AM
The arc does not not get drawn all the way

**Comment by andrewoid** [comment]
> The arc does not not get drawn all the way

And it looks like the circle isn't in the right spot.

**Comment by cblitz2** [comment]
I can't figure out why the arc isn't being drawn until the end

**Comment by andrewoid** [comment]
> I can't figure out why the arc isn't being drawn until the end

Maybe your for loop isn't going to the end the way you think it is.

**Comment by andrewoid** [comment]
replace the image

**Comment by cblitz2** [comment]
Screenshot 2024-02-27 at 3 14 53 PM

File: src/main/java/blitz/projectile/ProjectileGraph.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+        for (int i = 0; i < projectile.getApexTime() * 2 + 1; i++) {
+            currX = (int) projectile.getX();
+            currY = (int) projectile.getY();
+            projectile.setTime(i);
+            g.drawLine(currX, -currY, (int) projectile.getX(), -(int) projectile.getY());
+        }
+
+        g.setColor(Color.BLUE);
+        g.fillOval((int) projectile.getInterceptX() / 2, (int) -projectile.getPeakY(), 10, 10);
```
Comment (andrewoid) [inline src/main/java/blitz/projectile/ProjectileGraph.java]: the oval needs to be centered on the curve

**Comment by andrewoid** [comment]
Replace the screenshot in the description of the PR.

**Comment by cblitz2** [comment]
Screenshot 2024-02-28 at 4 27 18 PM

**Comment by andrewoid** [comment]
> Replace the screenshot in the description of the PR.

Read this comment again and follow what I said please.

**Review by andrewoid — APPROVED** [review]

**Pull request merged**